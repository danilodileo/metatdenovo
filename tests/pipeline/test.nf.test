nextflow_pipeline {
    name "Test pipeline"
    script "../../main.nf"
    tag "test"
    tag "PIPELINE"

    test("Test MEGAHIT/Prodigal") {

    when {
        params {
            outdir = "$outputDir"
            input = 'https://raw.githubusercontent.com/nf-core/test-datasets/metatdenovo/samplesheet/samplesheet.csv'
            assembler      = 'megahit'
            orf_caller     = 'prodigal'
            skip_eukulele  = true
            skip_eggnog    = true
            skip_kofamscan = true
        }
    }

    then {
        assertAll(
            { assert workflow.success },
            { assert new File("$outputDir/trimgalore/SAMPLE1_PE_1.fastq.gz_trimming_report.txt").exists() },
            { assert new File("$outputDir/trimgalore/fastqc/SAMPLE1_PE_1_val_1_fastqc.html").exists() },
            { assert snapshot(path("$outputDir/trimgalore/fastqc/SAMPLE1_PE_1_val_1_fastqc.html")).match("SAMPLE1_PE_1_val_1_fastqc.html") },
            { assert new File("$outputDir/megahit/megahit_out/megahit_assembly.log").exists() },
            { assert snapshot(path("$outputDir/megahit/megahit_out/megahit_assembly.log")).match("megahit_assembly.log") },
            { assert new File("$outputDir/prodigal/megahit.prodigal.gff.gz").exists() },
            { assert new File("$outputDir/samtools/SAMPLE1_PE.sorted.bam.bai").exists() },
            { assert new File("$outputDir/bbmap/logs/SAMPLE1_PE.bbmap.log").exists() },
            { assert snapshot(path("$outputDir/bbmap/logs/SAMPLE1_PE.bbmap.log")).match("SAMPLE1_PE.bbmap.log") },
            { assert new File("$outputDir/featurecounts/SAMPLE1_PE.featureCounts.tsv.summary").exists() },
            { assert snapshot(path("$outputDir/featurecounts/SAMPLE1_PE.featureCounts.tsv.summary")).match("SAMPLE1_PE.featureCounts.tsv.summary") },
            { assert new File("$outputDir/summary_tables/megahit.prodigal.counts.tsv.gz").exists() },
            { assert new File("$outputDir/summary_tables/megahit.prodigal.overall_stats.tsv.gz").exists() },
            { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
            { assert snapshot(path("$outputDir/multiqc/multiqc_report.html")).match("multiqc_report.html") },
            { assert new File("$outputDir/pipeline_info/nf_core_metatdenovo_software_mqc_versions.yml").exists() }
            )
        }
    }
}
