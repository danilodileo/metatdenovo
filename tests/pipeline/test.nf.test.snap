{
    "SAMPLE1_PE_1_val_1_fastqc.html": {
        "content": [
            "SAMPLE1_PE_1_val_1_fastqc.html:md5,89c71d5bf2499665b8c036a2ffae3bd7"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T17:26:13.683669"
    },
    "SAMPLE1_PE.bbmap.log": {
        "content": [
            "SAMPLE1_PE.bbmap.log:md5,63c03b3ac95d2f95cc0de3f8e2ac2548"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:38:50.513346"
    },
    "megahit_assembly.log": {
        "content": [
            "megahit_assembly.log:md5,536301548131f7424b156fa04f66822a"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:38:50.498041"
    },
    "multiqc_report.html": {
        "content": [
            "multiqc_report.html:md5,ecd4730774efa0856fb26ac6c3ef625c"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:38:50.573384"
    },
    "SAMPLE1_PE.featureCounts.tsv.summary": {
        "content": [
            "SAMPLE1_PE.featureCounts.tsv.summary:md5,55b0368c91e0007ec00a1e5352f0387c"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-19T20:38:50.524528"
    }
}